### PR TITLE
DEV: Refactor STI/polymorphic associations in chat

### PR DESCRIPTION
**Pull request by Flink**

### CODE REVIEW

File: plugins/chat/app/models/chat/message.rb [hunk 1]
```diff
@@ -78,24 +59,25 @@ def self.polymorphic_name
...
               },
             )
           }
-
     scope :created_before, ->(date) { where("chat_messages.created_at < ?", date) }
+    scope :uncooked, -> { where("cooked_version <> ? or cooked_version IS NULL", BAKED_VERSION) }
 
     before_save { ensure_last_editor_id }
 
+    def self.polymorphic_class_mapping = { "ChatMessage" => Chat::Message }
```
Comment (jjaffeux) [inline plugins/chat/app/models/chat/message.rb:79]: What your rule of thumb for prefixing with `Chat::` or not?

File: plugins/chat/app/models/concerns/chat/mappable.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      # the model used when loading chatable_type column
+      def polymorphic_class_for(name)
+        polymorphic_class_mapping[name] || super
+      end
+
+      # the chatable_type column value
+      def polymorphic_name
+        polymorphic_class_mapping.invert[self] || super
+      end
```
Comment (jjaffeux) [inline plugins/chat/app/models/concerns/chat/mappable.rb:29]: is `chatable_` in these comments still valid? looks to specific to me

File: plugins/chat/app/models/concerns/chat/mappable.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Chat
+  module Mappable
```
Comment (jjaffeux) [inline plugins/chat/app/models/concerns/chat/mappable.rb:4]: We should find a more descriptive name, anything can be "Mappable" maybe, `RemappablePolymorphism` or something like this...

File: plugins/chat/app/models/concerns/chat/mappable.rb (hunk 2 above)
Comment (Flink) [reply plugins/chat/app/models/concerns/chat/mappable.rb:29]: Indeed!

**Pull request merged**